opencog/attention/experimental/HebbianUpdatingAgent.cc

```diff
@@ -0,0 +1,155 @@
+/*
+ * opencog/attention/HebbianUpdatingAgent.cc
+ *
+ * Written by Roman Treutlein
+ * All Rights Reserved
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License v3 as
+ * published by the Free Software Foundation and including the exceptions
+ * at http://opencog.org/wiki/Licenses
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program; if not, write to:
+ * Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#include <algorithm>
+#include <math.h>
+#include <time.h>
+
+#include <opencog/util/Config.h>
+#include <opencog/util/mt19937ar.h>
+#include <opencog/attention/atom_types.h>
+
+#define DEPRECATED_ATOMSPACE_CALLS
+#include <opencog/atomspace/AtomSpace.h>
+#include <opencog/truthvalue/SimpleTruthValue.h>
+#include <opencog/atomutils/Neighbors.h>
+#include "HebbianUpdatingAgent.h"
+
+//#define DEBUG
+
+using namespace opencog;
+
+HebbianUpdatingAgent::HebbianUpdatingAgent(CogServer& cs) :
+        Agent(cs)
+{
+    // Provide a logger
+    log = NULL;
+    setLogger(new opencog::Logger("HebbianUpdatingAgent.log", Logger::FINE,
+    true));
+
+    as = &_cogserver.getAtomSpace();
+}
+
+HebbianUpdatingAgent::~HebbianUpdatingAgent()
+{
+    if (log)
+        delete log;
+}
+
+void HebbianUpdatingAgent::setLogger(Logger* _log)
+{
+    if (log)
+        delete log;
+    log = _log;
+}
+
+Logger* HebbianUpdatingAgent::getLogger()
+{
+    return log;
```

Comment:
you declare this log everywhere, but, as far as I can tell, you never actually use it. Do you even need the logger?